pyomo/contrib/doe/measurements.py

```diff
@@ -25,252 +25,270 @@
-
-    def _model_measure_name(self):
-        """Return pyomo string name
+        User can pass already defined measurement names here
         """
-        # store pyomo string name
-        measurement_names = []
-        # loop over measurement name
-        for mname in self.flatten_measure_name:
-            # check if there is extra index
-            if self.ind_string in mname:
-                measure_name = mname.split(self.ind_string)[0]
-                measure_index = mname.split(self.ind_string)[1]
-                for tim in self.flatten_measure_timeset[mname]:
-                    # get the measurement name in the model
-                    measurement_name = measure_name + '[0,' + measure_index + ',' + str(tim) + ']'
-                    measurement_names.append(measurement_name)
-            else:
-                for tim in self.flatten_measure_timeset[mname]:
-                    # get the measurement name in the model
-                    measurement_name = mname + '[0,' + str(tim) + ']'
-                    measurement_names.append(measurement_name)
-        self.model_measure_name = measurement_names
-
-    def SP_measure_name(self, j, t,scenario_all=None, p=None, mode='sequential_finite', legal_t=True):
-        """Return pyomo string name for different modes
-        Arguments
-        ---------
-        j: flatten measurement name
-        t: time 
-        scenario_all: all scenario object, only needed for simultaneous finite mode
-        p: parameter, only needed for simultaneous finite mode
-        mode: mode name, can be 'simultaneous_finite' or 'sequential_finite'
-        legal_t: if the time point is legal for this measurement. default is True
+        self.name = super().specify(self_define_res)
+
+        # generate default variance
+        self._use_identity_variance()
+
+    def add_elements(self, var_name, extra_index=None, time_index=[0]):
+        """
+        Parameters 
+        -----------
+        var_name: a ``list`` of measurement var names 
+            extra_index: a ``list`` containing extra indexes except for time indexes 
+                if default (None), no extra indexes needed for all var in var_name
+                if it is a nested list, it is a ``list`` of ``list`` of ``list``, 
+                they are different multiple sets of indexes for different var_name
+                for e.g., extra_index[0] are all indexes for var_name[0], extra_index[0][0] are the first index for var_name[0]
+            time_index: a ``list`` containing time indexes
+                default choice is [0], means this is an algebraic variable
+                if it is a nested list, it is a ``list`` of ``lists``, they are different time set for different var in var_name
+        """
+        self.name =  super().add_elements(var_name=var_name, extra_index=extra_index, time_index=time_index)
+
+        # generate default variance
+        self._use_identity_variance()
```

Comment:
Update this to rely on the user to specify an uncertainty (standard deviation) for each measurement. You can set the default to 1. Thus if the default is used, we will still get the identity covariance structure.